### PR TITLE
doc: add link to jenkins job from readme

### PR DESCRIPTION
**Pull request by Fishrock123**
Refs: https://github.com/nodejs/citgm/issues/137

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ node runtime.
 
 [![Build Status](https://travis-ci.org/nodejs/citgm.svg?branch=master)](https://travis-ci.org/nodejs/citgm)
 
+The deployed version of this, used by the Node.js Core team and hosted on multiple platforms, can be found [on our Jenkins CI](https://ci.nodejs.org/view/Node.js-citgm/job/thealphanerd-smoker/).
```
Comment (TheAlphaNerd) [inline README.md]: I don't know if `deployed` is the best term. We have a CI job set up to run this... but CITGM itself is just a node module, not a server.

Thoughts?

Comment (TheAlphaNerd) [inline README.md]: @Fishrock123  :D

**Comment by mhdawson** [comment]
lgtm to @TheAlphaNerd's version

**Comment by TheAlphaNerd** [comment]
landed in 1b7b13389c

**Pull request closed without merging**